esp/platform/espcfg.cpp

```diff
@@ -136,32 +136,16 @@ void CEspConfig::ensureSDSSessionDomains()
         Owned<IRemoteConnection> conn = querySDS().connect(PathSessionRoot, myProcessSession(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
         if (!conn)
         {
-            conn.setown(querySDS().connect("/", myProcessSession(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT));
-            if (!conn)
-                throw MakeStringException(-1, "Failed to connect SDS.");
-
-            IPropertyTree* sdsRoot = conn->queryRoot();
-            if (!sdsRoot)
-                throw MakeStringException(-1, "Failed to get SDS.");
-
-            Owned<IPropertyTree> sessionTree = createPTree();
+            conn.setown(querySDS().connect("/", myProcessSession(), RTM_LOCK_WRITE|RTM_CREATE_QUERY, SESSION_SDS_LOCK_TIMEOUT));
+            IPropertyTree* sessionTree = ensurePTree(conn->queryRoot(), PathSessionRoot);
```

Comment:
If the PathSessionRoot already exists, will the RTM_CREATE_QUERY overwrite the existing one with a new one?